### ISSUE TITLE
problem with loading core fontello font with static 

### ISSUE THREAD

**Issue by Jimmi08** [e107inc/e107#4358]
I use e_HTTP_STATIC for my site. Everything works except the core font.

`Access to font at 'https://static.e107sk.com/e107_plugins/social/font/fontello.woff2?21917124' from origin 'https://www.e107sk.com' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.`

But I have listed it in htaccess
```
<FilesMatch "\.(js|css|ico|pdf|flv|jpg|jpeg|png|gif|swf|mp3|mp4|eot|otf|ttc|ttf|woff|woff2)$">
    Header set Cache-Control "public"
    Header unset Cookie
    Header unset Set-Cookie
    Header append Vary: Accept-Encoding
    Header set Access-Control-Allow-Origin "https://www.e107sk.com"
</FilesMatch>
```

Any ideas what is wrong? And only with one file?  Sometimes font icons are not visible.

Thanks

**Comment by tgtje** [comment]
key = Access-Control-Allow-Origin  have seen this issue elsewhere ( where?) Is related to the stricter rules being applied in browser around last November/December  [(info)](https://developer.mozilla.org/nl/docs/Web/HTTP/CORS). hope this might help in a way, if this would be applicable to the issue.

**Comment by Jimmi08** [comment]
Yes, you need to add an exception to your htaccess... but it is there and it doesn't work only for this one font.  Check the console on my site if you want.  On news item view- the email icon is not visible sometimes. 

That regex is probably wrong, it works for woff2  but it is woff2?21917124

**Comment by Jimmi08** [comment]
It is bug.

This line is causing this:

`e107::link(array('rel'=>'preload', 'href'=> "{e_PLUGIN}social/font/fontello.woff2?21917124", 'as' => "font", 'type'=>"font/woff2", 'crossorigin'   => true));`

After removing this line, everything is correct.

**Comment by tgtje** [comment]
Now i do not always get the meaning of ht rules and conditions  ;-) , 
But for testing why not this? (borrowed from another site)

`<FilesMatch ".(eot|otf|ttf|woff|woff2)">
    Header always set Access-Control-Allow-Origin "*"
</FilesMatch>`

So test with only fonts rule.?

But finding similar patterns, fonts are in front and the second line with the * is often used example : 
`<IfModule mod_headers.c>
    <FilesMatch "\.(ttf|ttc|otf|eot|woff|font.css|css|js|gif|png|jpe?g|svg|svgz|ico|webp)$">
        Header set Access-Control-Allow-Origin "*"
    </FilesMatch>
</IfModule>`
not sure here iff i am helping, but hey _noticed... webp is recently introduced_, but is not in the org. posted ;-).
Because of the findings, it seems to happen more on fontello one might conclude.
As for the 'bug line' seems following the rules of w3c related to cors, but crossorigin' => true  (again -  knowledge) true is imo already ruled by htaccess, or i misunderstand..)(iff : sorry )

**Comment by CaMer0n** [comment]
> e107::link(array('rel'=>'preload', 'href'=> "{e_PLUGIN}social/font/fontello.woff2?21917124", 'as' => "font", 'type'=>"font/woff2", 'crossorigin' => true));
> 
> After removing this line, everything is correct.

It removes the 'bug', but it also removes the preloading. 

What if you remove `$` from this line?

`<FilesMatch "\.(js|css|ico|pdf|flv|jpg|jpeg|png|gif|swf|mp3|mp4|eot|otf|ttc|ttf|woff|woff2)$">`

**Comment by Jimmi08** [comment]
> What if you remove `$` from this line?
> 
> `<FilesMatch "\.(js|css|ico|pdf|flv|jpg|jpeg|png|gif|swf|mp3|mp4|eot|otf|ttc|ttf|woff|woff2)$">`

That example is in e107.htaccess, I tried, 500 error

> It removes the 'bug', but it also removes the preloading.
that preloading is wrong in my case too, I am getting a warning (not error) from the console about not correct using it. You can see it on e107.org

`The resource https://static.e107.org/e107_web/lib/font-awesome/4.7.0/css/font-awesome.min.css?1611626819 was preloaded using link preload but not used within a few seconds from the window's load event. Please make sure it has an appropriate `as` value and it is preloaded intentionally.`

**Comment by Jimmi08** [comment]
@Moc Could you remove label question please? Thanks
